### ISSUE TITLE
Godot animations are extremely hard to use

### ISSUE THREAD

**Issue by slapin** [godotengine/godot#2246]
- [ ] AnimationTreePlayer Filter mechanism is counter-intuitive and buggy for Blend nodes.
  It is very hard to find the intended track as bone hierarchy doesn't really work there and
  very hard to figure-out what filters actually do.
- [X] AnimationTreePlayer totally fail with non-looping animations (it can play them only once, then
  only last frame is played)
- [ ] Importer imports non-deform bones even if told not to, which adds hundreds of unneeded tracks
  on complex rigs
- [ ] AnimationTreePlayer is very hard to use with animation count > 20, as it is not possible to group
  nodes, hide/show nodes, etc. It is very hard to visually find needed animations.

**Comment by slapin** [comment]
I've tested 1.0 stable, animations behave the same, so non-looping animations are never actually worked.

**Comment by mamarilmanson** [comment]
just curious... is this 3d? :)

**Comment by slapin** [comment]
yes, but AnimationTreePlayer is not really 3d-specific, but it is hard to
use with 2d anyway

**Comment by mamarilmanson** [comment]
oh... I get it... :)

**Comment by akien-mga** [comment]
Again, it's hard to work with so many points in one issue, especially since some of them seem to be subjective ("extremely hard to use"...). Please open atomic issues that really pinpoint a problem that can be fixed, and not just "this sucks because A, B, C, D and E reasons.".

**Comment by slapin** [comment]
Well, IMHO the way this issue is closed looks like "nobody cares" reason.
And there is nothing subjective in this report, the problems are well known
and was reported by many people.

**Comment by akien-mga** [comment]
Quoting:

> it's hard to work with so many points in one issue

> Please open atomic issues that really pinpoint a problem that can be fixed

Do what you want with that.

**Comment by slapin** [comment]
You didn't even try to understand this issue. Are you hunting for closed
bugs number? A hint: You can open issues and close them, that will increase
count.

This bug is for AnimationTreePlayer. You never used it and you will never
understand it. So don't touch it.

**Comment by akien-mga** [comment]
As you wish, but don't come whining if this bug gets no further attention in the next two years. I'm just trying to help, but if you don't think bug triage is relevant, it's fine by me, I won't touch your issues anymore.

**Comment by slapin** [comment]
Please consider this bug as reminder for me. As it seems nobody else really cares for problems mentioned here. I need to debug what is going on, but have no much time as anybody else.
As soon as I will have deeper understanding of the problem I will come up with proper bugreport(s).
If you close it now it will definitely be forgotten and have no attention next N years.

**Comment by vnen** [comment]
> Are you hunting for closed bugs number?

He's actually asking you to open one new issue for each of your points, which would increase the number of open bugs.

The problem here is that this is covering too much ground in a single report, which makes hard to work with. Say someone wants to do this and claim this issue, they'll have to solve every cited problem, which they might not have the knowledge to do. So people won't be willing to claim this. If it were only on single point, there's a higher chance of someone in the community accepts to solve it.

Also, note that you're 3rd point is not even about the AnimationTreePlayer, but about the importer, which is a different category. Since we will be handling issues for teams of specific areas, we will have to tag a lot of teams in this single issue and they'll have to dig the discussion to see what actually needs to be done by them and what is for another team.

Just wanted to make that clear.

**Comment by slapin** [comment]
Sorry folks, I really know what I'm doing here - this report is dedicated
to AnimationTreePlayer.
As soon as I get to the sight of things, I will make better report(s). That
will take some time.
For the time being please leave this one open. If you can, however, I'm not
against splitting this one, but
it should be labelled somehow to be distinguished as the one.
As for 3. - my bad, but still I found nothing like this reported, can be
considered feature request or removed. But I need to be able to easily find
AnimationTreePlayer stuff - in this case I'll be fine with it.

**Comment by reduz** [comment]
I don't mind this issue, it eventually has to be fixed

**Comment by reduz** [comment]
there is not going to be serious work from my part on this until 3.0

**Comment by reduz** [comment]
kicked to 3.1, ran out of time on this
